From 05c7231276b36973447cf58c11d7377c1d98217c Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Tue, 27 Mar 2018 19:06:48 +0800
Subject: [PATCH 1362/2302] [INTERNAL]drm/i915/gvt: port A HDMI emulation is
 only for BXT platform

KBL has more ports to support 3 HDMI monitors, then don't need
force initializing HDMI for port A.
Guest OS only patch.

Change-Id: Ieae488d242d3600773ab18194ac18a5097efe4b2
Signed-off-by: Fei Jiang <fei.jiang@intel.com>
---
 drivers/gpu/drm/i915/intel_ddi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ddi.c b/drivers/gpu/drm/i915/intel_ddi.c
index b233723..1325a5b 100644
--- a/drivers/gpu/drm/i915/intel_ddi.c
+++ b/drivers/gpu/drm/i915/intel_ddi.c
@@ -2796,8 +2796,8 @@ void intel_ddi_init(struct drm_i915_private *dev_priv, enum port port)
 
 	/* In theory we don't need the encoder->type check, but leave it just in
 	 * case we have some really bad VBTs... */
-	if ((intel_vgpu_active(dev_priv) ||
-		(intel_encoder->type != INTEL_OUTPUT_EDP && init_hdmi))) {
+	if ((intel_vgpu_active(dev_priv) && IS_BROXTON(dev_priv)) ||
+		(intel_encoder->type != INTEL_OUTPUT_EDP && init_hdmi)) {
 		if (!intel_ddi_init_hdmi_connector(intel_dig_port))
 			goto err;
 	}
-- 
2.7.4

